keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,396 @@
+
+#### cAdvisor
+
+Once CRI and Kubelet stats provider level changes are in place, we can evaluate disabling cAdvisor from collecting container and pod level stats. We may need to introduce new ability in cAdvisor to disable on-demand collection for certain cgroup hierarchies to ensure we can continue using cAdvisor for only node/machine level stats. 
+
+### User Stories [optional]
+
+#### Story 1
+As a cluster admin, I would like my node to be as performant as possible, and not waste resources by duplicating stats collection by CRI and cAdvisor.
+
+#### Story 2
+As a node maintainer, I would like to clarify the sources of truth for each stat the Kubelet reports to the Summary API, so that I can better find which component is to investigate for any issues.
+
+
+### Risks and Mitigations
+
+- To properly move to CRI stats, it is likely there are some metrics that we'll want to not support (Accelerator/UserDefined). We should be careful to not break entities that rely on these metrics.
+- A large part of this work is changing the source of the Summary API metrics. In doing so, there is a risk that collecting from a new source will change how the stats look in aggregate (and risk bugs popping up in new areas).
+- cAdvisor has a long history of collecting these stats. There is a risk that changing the source of the stats to the CRI implementation can cause performance regressions
+as cAdvisor is fine tuned to perform in an adequate manner.
+	- CRI implementations should do performance regression analyses to ensure the change does not regress too much.
+
+## Graduation Criteria
+
+
+## Implementation History
+
+
+## Drawbacks [optional]
+
+CRI runtimes will each have to implement additional interface to support full stats, rather than all metric collection being unified by cAdvisor. 
+
+Note: This is by design as this will enable to decouple runtime implementation details further from kubelet.
+
+## Alternatives [optional]
+
+Instead of teaching CRI how to do *everything* cAdvisor does, we could instead have cAdvisor not do the work the CRI stats end up doing (specifically when reporting disk stats, which are the most expensive operation to report). However, this doesn't address the anti-pattern of having multiple parties confusingly responsible for a wide array of metrics and other issues described.
+
+### Notes [optional]
+
+**TODOs**:
+
+* The fields in PodSandboxStats will largely mirror ContainerStats, any room to simplify that?
+* Should we support [User Defined metrics](https://github.com/google/cadvisor/blob/master/docs/application_metrics.md)? This feature is alpha and does not appear to be widely used?
+* There are some fields that are defined in SummaryAPI (e.g time in a lot). Do we add those?
+* How do we tell cAdvisor to not collect container metrics?
+* - Give it a bogus root?
+* - Flag?
```

Comment:
How about a configuration file as a third option?

Reply:
I can't find evidence there is a config file for cAdvisor. @bobbypage is this an option?

Reply:
I added "cAdvisor configuration option", which could mean config file if possible, or the config struct used in cAdvisor startup

Reply:
I don't think cAdvisor supports file based config currently. Adding something like that may make sense, or maybe just adding a flag if needed similar to how kubelet configures cAdvisor options currently:  https://github.com/kubernetes/kubernetes/blob/5d6dc8d/pkg/kubelet/cadvisor/cadvisor_linux.go#L70-L76